softwareconfiguration/model/src/main/java/ai/libs/softwareconfiguration/model/ComponentInstance.java

```diff
@@ -1,329 +1,327 @@
+			if (other.component != null) {
+				return false;
+			}
+		} else if (!this.component.equals(other.component)) {
+			return false;
+		}
+		if (this.parameterValues == null) {
+			if (other.parameterValues != null) {
+				return false;
+			}
+		} else if (!this.parameterValues.equals(other.parameterValues)) {
+			return false;
+		}
+		if (this.satisfactionOfRequiredInterfaces == null) {
+			if (other.satisfactionOfRequiredInterfaces != null) {
+				return false;
+			}
+		} else if (!this.satisfactionOfRequiredInterfaces.equals(other.satisfactionOfRequiredInterfaces)) {
+			return false;
+		}
+		return true;
+	}
+
+	public String toComponentNameString() {
+		StringBuilder sb = new StringBuilder();
+		sb.append(this.getComponent().getName());
+		if (!this.satisfactionOfRequiredInterfaces.isEmpty()) {
+			sb.append(this.satisfactionOfRequiredInterfaces.entrySet().stream().map(x -> x.getValue().toComponentNameString()).collect(Collectors.toList()).toString());
+		}
+		return sb.toString();
+	}
+
+	@JsonIgnore
+	@Override
+	public String toString() {
+		Map<String, Object> fields = new HashMap<>();
+		fields.put("component", this.component);
+		fields.put("parameterValues", this.parameterValues);
+		fields.put("satisfactionOfRequiredInterfaces", this.satisfactionOfRequiredInterfaces);
+		return ToJSONStringUtil.toJSONString(this.getClass().getSimpleName(), fields);
+	}
+
+	/**
+	 * Returns the description of a <code>ComponentInstance</code> as a pretty print with indentation.
+	 *
+	 * @return A string representing this object in JSON format.
+	 * @throws IOException
+	 *             An IOException is thrown if the object cannot be serialized to a String.
+	 */
+	@JsonIgnore
+	public String getPrettyPrint() throws IOException {
+		return new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).writeValueAsString(this);
+	}
+
+	public String getNestedComponentDescription() {
+		StringBuilder sb = new StringBuilder();
+		sb.append(this.getComponent().getName());
+		this.satisfactionOfRequiredInterfaces.values().stream().map(x -> " - " + x.getNestedComponentDescription()).forEach(sb::append);
+		return sb.toString();
+	}
+
+	/**
+	 * Add an annotation to this component instance.
+	 * @param key The key of how to address this annotation.
+	 * @param annotation The annotation value.
+	 */
+	public void putAnnotation(final String key, final String annotation) {
+		this.annotations.put(key, annotation);
+	}
+
+	/**
+	 * Retrieve an annotation by its key.
+	 * @param key The key for which to retrieve the annotation.
+	 * @return The annotation value.
+	 */
+	public String getAnnotation(final String key) {
+		return this.annotations.get(key);
+	}
+
+	public void appendAnnotation(final String key, final String annotation) {
+		if (this.annotations.containsKey(key)) {
+			this.annotations.put(key, this.annotations.get(key) + annotation);
+		} else {
+			this.annotations.put(key, annotation);
+		}
+	}
+
+	public boolean isDefaultParametrized() {
```

Comment:
Do we really want to have this method in a ComponentInstance or does this rather belong to a util class?